src/Workspaces/Core/Portable/Workspace/FileTextLoader.cs

```diff
@@ -58,18 +58,29 @@ public FileTextLoader(string path, Encoding? defaultEncoding)
 
         internal sealed override string FilePath => Path;
 
-        protected virtual SourceText CreateText(Stream stream, Workspace workspace)
-        {
-            var factory = workspace.Services.GetRequiredService<ITextFactoryService>();
-            return factory.CreateText(stream, DefaultEncoding);
-        }
+        /// <summary>
+        /// Creates <see cref="SourceText"/> from <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">Stream.</param>
+        /// <param name="workspace">Obsolete. Null.</param>
+        [Obsolete("Use CreateText(Stream, CancellationToken)")]
+        protected virtual SourceText CreateText(Stream stream, Workspace? workspace)
```

Comment:
Oh this is actually kinda nifty -- even if somebody is using this obsolete method then the nullable annotation makes it clear what the change was.